File: src/org/symbolmath/util/SymbolMathUtil.java
package org.symbolmath.util;

import org.symbolmath.ast.ASTElement;

public class SymbolMathUtil {

  public static void checkParents(ASTElement element, ASTElement parent) {
    if (element.getParent() != parent) {
      throw new RuntimeException("Wrong parent");
    }
    for (ASTElement child : element.getChildren()) {
      checkParents(child, element);
    }
  }

  public static boolean isIdentifier(String s) {
    if (!Character.isJavaIdentifierStart(s.charAt(0))) {
      return false;
    }
    for (int i = 1; i < s.length(); i++) {
      if (!Character.isJavaIdentifierPart(s.charAt(i))) {
        return false;
      }
    }
    return true;
  }

  public static Integer parseAsInt(String s) {
    try {
      return Integer.parseInt(s);
    } catch (NumberFormatException e) {
      return null;
    }
  }

  public static Double parseAsFloat(String s) {
    try {
      return Double.parseDouble(s);
    } catch (NumberFormatException e) {
      return null;
    }
  }

}

File: src/main/java/org/symbolmath/util/SymbolMathUtil.kt
package org.symbolmath.util

import org.symbolmath.ast.ASTElement
import java.lang.Character.*

object SymbolMathUtil {

    fun checkParents(element: ASTElement, parent: ASTElement?) {
        if (element.parent !== parent) {
            throw RuntimeException("Wrong parent")
        }
        for (child in element.children) {
            checkParents(child, element)
        }
    }

    fun isIdentifier(s: String): Boolean {
        if (!isJavaIdentifierStart(s[0])) {
            return false
        }
        for (c in s) {
            if (!isJavaIdentifierPart(c)) {
                return false
            }
        }
        return true
    }

    @JvmStatic
    fun parseAsInt(s: String): Int? {
        return try {
            s.toInt()
        } catch (e: NumberFormatException) {
            null
        }
    }

    fun parseAsFloat(s: String): Double? {
        return try {
            s.toDouble()
        } catch (e: NumberFormatException) {
            null
        }
    }
}